co-authors-plus.php

```diff
@@ -1133,6 +1133,12 @@ function current_user_can_set_authors( $post = null ) {
 	 */
 	public function fix_author_page( $selection ) {
 
+		global $wp_query, $authordata;
+ 
+		if ( ! isset( $wp_query ) ) {
+			return;
```

Comment:
I monkey-patched my instance with this, and found it doesn't seem to help.

At least in our case, `$wp_query` is always set here, so this check does nothing to avoid executing further inappropriate code :thinking:

Reply:
@lkraav Can you tell me for which plugin you're seeing the error message, so that I can reproduce your case on my end?

In addition, I'll also try the solution that you suggested on https://github.com/Automattic/Co-Authors-Plus/issues/615#issuecomment-855820522.

Reply:
WCM === WooCommerce Memberships

Do you have a access to Woo commercial plugin copies?

Reply:
I do have access to the commercial WooCommerce plugins and strange enough, I do not see the problem appearing when using [WooCommerce Memberships](https://woocommerce.com/products/woocommerce-memberships/). That said, I've updated the PR using your code. That fix works fine on my end and I assume it works fine on your end as well. Can you confirm that? And can you confirm that we're talking about the same plugin?

Reply:
> I do have access to the commercial WooCommerce plugins and strange enough, I do not see the problem appearing when using WooCommerce Memberships.

Sentry traceback I posted appears like a generic bootstrap process, but traceback public view doesn't fails to show this particular event sample was triggered by a POST request, which may have some different code paths from GET requests.

WCM also caches membership post query results, so only initial request for that user may be triggering it, etc.

> Can you confirm that?

Everything confirmed. Proof from live server - middle box is 24h event bar chart, and you can see it die down immediately after monkey-patching with `did_action wp`.

![image](https://user-images.githubusercontent.com/147228/121149720-92725180-c84b-11eb-8d79-7af08626a8ee.png)

One remaining thing perhaps: I have not personally verified anything about what this filter is originally meant to achieve in the first place - whether that functionality still works as intended. I would assume you've been able to take a look?

Reply:
PS I've been using `did_action()` timing strategy successfully elsewhere as well, and also seen it done in the ecosystem. It's not a random idea.

Reply:
> One remaining thing perhaps: I have not personally verified anything about what this filter is originally meant to achieve in the first place - whether that functionality still works as intended. I would assume you've been able to take a look?

Pippin posted a great explanation of what `did_action` is doing on https://pippinsplugins.com/detect-when-an-action-has-run-with-did_action/. That filter was new to me, and it simply counts how often a certain function was called. This is useful when creating CPT's to ensure that they're only created during the first run. In this particular case, we do not execute the `is_author` function if the `wp` action hasn't been executed yet. 

Regarding the functionality, all PHPUnit tests are running as expected, and I did not spot any problems after adding `did_action`.

Reply:
Great to hear, but I should've been more clear in my comment: concern was about original `posts_selection` action, whose logic we're modifying here.

Unit tests passing sounds good to me, didn't think to check for their existence in this repo.

Reply:
> At least in our case, $wp_query is always set here, so this check does nothing to avoid executing further inappropriate code 🤔

@lkraav Can you please tell me what the value of `$wp_query `is when it is set?

I feel like the `isset()` check for `$wp_query` is headed in the right direction. However, instead of checking if it's set, perhaps we can more explicitly check what the function expects `$wp_query` to be.

Reply:
> At least in our case, $wp_query is always set here, so this check does nothing to avoid executing further inappropriate code 🤔

@lkraav Can you please tell me what the value of `$wp_query `is when it is set? Is it a NULL or empty value?